templates/reservation/reservation.py

```diff
@@ -30,9 +32,22 @@ def __init__(self, name, guid=None, data=None):
     def validate(self):
         if not self.data.get('creationTimestamp'):
             self.data['creationTimestamp'] = time.time()
+        if not self.data.get('expiryTimestamp'):
+            self.data['expiryTimestamp'] = time.time()
+
+    def extend(self, duration):
+        expiry = datetime.fromtimestamp(self.data["expiryTimestamp"])
+        month = expiry.month - 1 + duration
+        year = expiry.year + month // 12
+        month = month % 12 + 1
+        day = min(expiry.day, calendar.monthrange(year,month)[1])
+        extended = datetime(year, month, day, expiry.hour, expiry.minute, expiry.second, expiry.microsecond)
+        self.data["expiryTimestamp"] = int(time.mktime(extended.timetuple()))
+
+        return self.data["expiryTimestamp"]
 
     @retry((Exception), tries=4, delay=5, backoff=2, logger=None)
-    def install(self):
+    def install(self,duration):
```

Comment:
This is very unusuall to have a argument with the `install` action, why is this required ?

Reply:
Yea, actually I can just compute it in the grid_broker and set it in the data.

Reply:
don't forget to remove it